File: app/src/main/java/com/commit451/gitlab/model/api/CommitUser.java
package com.commit451.gitlab.model.api;

import com.commit451.gitlab.util.ObjectUtil;
import com.squareup.moshi.Json;

import org.parceler.Parcel;

@Parcel
public class CommitUser {
    @Json(name = "id")
    String id;
    @Json(name = "name")
    String name;
    @Json(name = "username")
    String username;
    @Json(name = "state")
    String state;
    @Json(name = "avatar_url")
    String avatarUrl;
    @Json(name = "web_url")
    String webUrl;

    public CommitUser() {}

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getUsername() {
        return username;
    }

    public String getState() {
        return state;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public String getWebUrl() {
        return webUrl;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof CommitUser)) {
            return false;
        }

        CommitUser commit = (CommitUser) o;
        return ObjectUtil.INSTANCE.equals(id, commit.id);
    }

    @Override
    public int hashCode() {
        return ObjectUtil.INSTANCE.hash(id);
    }
}

File: app/src/main/java/com/commit451/gitlab/model/api/CommitUser.kt
package com.commit451.gitlab.model.api

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
data class CommitUser(
    @Json(name = "id")
    var id: String? = null,
    @Json(name = "name")
    var name: String? = null,
    @Json(name = "username")
    var username: String? = null,
    @Json(name = "state")
    var state: String? = null,
    @Json(name = "avatar_url")
    var avatarUrl: String? = null,
    @Json(name = "web_url")
    var webUrl: String? = null
) : Parcelable
